cli/tests/pcluster/validators/test_instances_validators.py

```diff
@@ -476,25 +476,48 @@ def test_instances_networking_validator(
 
 
 @pytest.mark.parametrize(
-    "compute_resource_name, capacity_type, allocation_strategy, expected_message",
+    "capacity_type, allocation_strategy, expected_message",
     [
         # OnDemand Capacity type only supports "lowest-price" allocation strategy
-        # Spot Capacity type supports both "lowest-price" and "capacity-optimized" allocation strategy
         (
-            "TestComputeResource",
             CapacityType.ONDEMAND,
             AllocationStrategy.CAPACITY_OPTIMIZED,
             "Compute Resource TestComputeResource is using an OnDemand CapacityType but the Allocation Strategy "
             "specified is capacity-optimized. OnDemand CapacityType can only use 'lowest-price' allocation strategy.",
         ),
-        ("TestComputeResource", CapacityType.ONDEMAND, AllocationStrategy.LOWEST_PRICE, ""),
+        (CapacityType.ONDEMAND, AllocationStrategy.LOWEST_PRICE, ""),
+        (
+            CapacityType.ONDEMAND,
+            None,
+            "Compute Resource TestComputeResource is using an OnDemand CapacityType but the Allocation Strategy "
+            "specified is not set. OnDemand CapacityType can only use 'lowest-price' allocation strategy.",
+        ),
+        # Spot Capacity type supports both "lowest-price" and "capacity-optimized" allocation strategy
+        (CapacityType.SPOT, AllocationStrategy.LOWEST_PRICE, ""),
+        (CapacityType.SPOT, AllocationStrategy.CAPACITY_OPTIMIZED, ""),
+        (CapacityType.SPOT, None, ""),
```

Comment:
why this combination isn't generating a message like in the case for CapacityType.ONDEMAND / None ?

Reply:
Modified validator to not show the message in the on-demand case when the value is None.

This is an internal change only , because allocations-strategy has always a default value.